lib/Bio/EnsEMBL/DataCheck/Checks/ExonsBoundaries.pm

```diff
@@ -0,0 +1,163 @@
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::ExonsBoundaries;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use List::Util qw[min max];
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+    NAME        => 'ExonsBoundaries',
+    DESCRIPTION => 'Checks exon boundaries in their transcript (same strand / correct order / min start / max end)',
+    DB_TYPES    => [ 'core' ],
+    GROUPS      => [ 'core_handover' ],
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $mapper = sub {
+    my ($row, $value) = @_;
+    my %row = (
+        gene_id           => $$row[0],
+        transcript_id     => $$row[1],
+        transcript_strand => $$row[2],
+        transcript_start  => $$row[3],
+        transcript_end    => $$row[4],
+        exon_id           => $$row[5],
+        exon_start        => $$row[6],
+        exon_end          => $$row[7],
+        exon_strand       => $$row[8],
+        exon_rank         => $$row[9]
+    );
+    return \%row;
+  };
+
+  my $sql_strand_order = q/
+    SELECT  g.gene_id,
+            tr.transcript_id,
+            tr.seq_region_strand,
+            tr.seq_region_start,
+            tr.seq_region_end,
+            e.exon_id,
+            e.seq_region_start,
+            e.seq_region_end,
+            e.seq_region_strand,
+            et.`rank`
+    FROM    gene g
+    INNER JOIN transcript tr USING (gene_id)
+    INNER JOIN exon_transcript et  USING (transcript_id)
+    INNER JOIN exon e USING (exon_id)
+    WHERE  tr.transcript_id NOT IN (
+	          SELECT  transcript_id
+	          FROM    transcript_attrib
+	          INNER JOIN attrib_type
+	          USING (attrib_type_id)
+	          WHERE code='trans_spliced')
+    ORDER BY et.transcript_id, et.`rank`
+    /;
+
+
+  my $lastTranscriptID = -1;
+  my $lastExonStart = -1;
+  my $lastExonEnd = -1;
+  my $lastExonStrand = -2;
+  my $lastExonID = -1;
+  my $lastExonRank = 0;
+  my $minExonStart = -1;
+  my $maxExonEnd = 0;
+  my $lastTranscriptStart = -1;
+  my $lastTranscriptEnd = 0;
+
+  my @genes_exons_strand = @{$self->dba->dbc->sql_helper->execute_simple(-SQL => $sql_strand_order, -CALLBACK => $mapper)};
+
+  my @exons_not_on_same_strands;
+  my @exon_transcript_different_strand;
+  my @exons_overlaps;
+  my @exon_transcript_start_end;
+
+  foreach my $strand (@genes_exons_strand) {
+    if ($$strand{transcript_id} == $lastTranscriptID) {
+      if ($lastExonStrand < -1) {
+        $lastExonStrand = $$strand{exon_strand};
+        $lastExonStart = $$strand{exon_start};
+        $lastExonEnd = $$strand{exon_end};
+        $lastExonID = $$strand{exon_id};
+        $lastExonRank = $$strand{exon_rank};
+        $minExonStart = $$strand{transcript_start};
+        $maxExonEnd = $$strand{transcript_end};
+        $lastTranscriptEnd = $$strand{transcript_end};
+        $lastTranscriptStart = $$strand{transcript_start};
+      }
+      else {
+        if ($$strand{exon_strand} != $lastExonStrand) {
```

Comment:
This test isn't necessary. We have a test to check whether exon and transcript are different. By definition, for a multi-exon transcript, if two exons have different strands, then one of them will be different to the transcript; so checking whether exons have different strands is redundant.